genvalidity/src/Data/GenValidity.hs

```diff
@@ -118,6 +148,20 @@ import Control.Monad (forM)
 --
 -- > instance GenUnchecked MyType where
 -- >     genUnchecked = MyType <$> genUnchecked <*> genUnchecked
+--
+--
+-- __Step 2__: If an instatiation via 'Generic' is not possible, then you should emulate what
+--         'genericGenUnchecked' does.
+--         This means that all sub-parts should be  generated using 'genUnchecked'.
+--         Make sure to generate any possible value, valid or not, that can exist at runtime
+--         even when taking the existance of 'unsafeCoerce' into account.
```

Comment:
Because it is not in scope, it will not be linked, you wanna write: `'Unsafe.Coerce.unsafeCoerce'`. 

Here is my opinion on the subject:

As a library developer I can prevent __some__ values from coming into existence for a particular data type, by utilizing the biggest power that Haskell has, its type system. When I know that certain invalid values cannot be created, I can use that invariant as a precondition to all of my functions. By dropping extra validations at runtime, I improve performance of my library without sacrificing its safety.

Taking such data type as an example, I see three problems with you rules that the user of genvalidity must create all possible values with `GenUnchecked` (consequently also `GenInvalid`) even by such extreme means as using `unsafeCoerce` or digging into internal modules that are considered unsafe (such as `Data.ByteString.Internal`, for instance):
1. You can no longer apply randomly generated data to your functions, because their precondition is violated, hence they can fail in arbitrary ways, possibly even with segfaults.
2. You can no longer test your functions for invalid data that can potentially be constructed, thus making `GenUnchecked` useless for such data types.
3. `GenUnchecked` generator, although exists, cannot be used for deriving such instances for other types that are using the data type.

Reply:
Hi @lehins. 
This is a super interesting topic to me so I hope you'll have the patience to point out the things that I miss.

Before I go on: If you need valid `ByteString` values, use `genValid`, not `genUnchecked`.
Use `genValidRecursively` if your types contain a `ByteString`.

What you seem to suggest is a different semantics for 'genUnchecked'. Let's talk about that.
The current semantics fo 'genUnchecked' are the only one that allow for automatic implementation via 'Generic'.
The idea is that if 'ByteString' had a 'Generic' instance, and derived the 'GenUnchecked' instance, then the instance should still be correct.
Now, 'ByteString' is a very dangerous type, so that results in all sorts of madness.

Now, there are other things we can think about:
Maybe `genUnchecked` for `ByteString` should be `error "you wouldn't want to use this anyway, use genValid if you want valid ByteString values and use genValidRecursively if your type contains a ByteString"`
Maybe the `genUnchecked` instance for `ByteString` should be unlawful and just implemented as `SB.pack <$> genUnchecked`.
These are things to think about.
Whatever we decide, I need to make sure that `genvalidity-text` is also changed accordingly, and I will write up a document about it like I did for `DOUBLE.md`.

Reply:
I guess there are a few things  I am suggesting:

* `GenInvalid`, `GenUnchecked` and `GenValid` are classes that a user suppose to implement for their data types. So this is not category theory, those type classes are used for testing, so there there should be no "laws", but suggestions on how to use them properly. It should not be that  `Generic` like derivation is used as the source of truth, but a user's perspective of what is "valid", "invalid" or "unchecked" is.
* Considering the idea above, `SB.pack <$> genUnchecked` implementation for `ByteString` is no longer "unlawful" and absolutely fine. It's only fine, it is actually useful

You probably disagree with me on the above suggestions, so the next best things I can recommend to deal with this are:
* to have an instance that throws an error instead of producing the code that segfaults
* or at least add a `{-# WARNING genUnchecked "Don't use this, it is utterly broken" #-}`

I guess that is all I have to say about this, so it's up to you however you wanna proceed with this